src/app/auth/services/auth.service.ts

```diff
@@ -322,11 +322,24 @@ export class AuthService {
    */
   async guardAuthenticated(): Promise<boolean> {
     const state = await this.#determineSessionState();
-    if (state == 'Authenticated') return true;
+    if (state === 'Authenticated') return true;
     this.#notify.showWarning('Please login to continue the requested action.');
     return this.#guardBack();
   }
 
+  /**
+   * This method can be used as a guard for all routes available only to data stewards.
+   * @returns true if the route is accessible
+   */
+  async guardDataSteward(): Promise<boolean> {
+    const state = await this.#determineSessionState();
+    if (state === 'Authenticated' && this.role() === 'data_steward') return true;
+    this.#notify.showWarning(
+      'Please login as data steward to continue the requested action.',
```

Comment:
I feel like this is too much information to display for non-data stewards. Why not 403 or 404? I know security by obfuscation is bad practice, especially when the website code is FOSS, but still.

Reply:
The problem appears when you bookmark the page and then visit it without being logged in as a data steward.

Maybe just "Insufficient permissions for the requested action?"

Reply:
Perhaps. It would be less stark than the 403 for sure.

Reply:
Changed in a5452e5. Does not mention the data steward role any more, but makes distinction between being logged in and logged out to provide more appropriate message.